core/src/main/java/io/questdb/griffin/engine/groupby/CountRecordCursorFactory.java

```diff
@@ -100,10 +100,8 @@ public boolean hasNext() {
                 } else {
                     count = 0;
                     try {
-                        while (baseCursor.hasNext()) {
-                            circuitBreaker.statefulThrowExceptionIfTripped();
-                            count++;
-                        }
+                        count = baseCursor.calculateSize(circuitBreaker);
+                        circuitBreaker.statefulThrowExceptionIfTripped();
```

Comment:
do we have to check the circuit breaker here?

Reply:
I left it because factories using async filter behave a bit different regarding circuit breaker check and without this line a pg cancellation test started failing .  Let me check the test again .

Reply:
I added timeout check to async filter cursor's method and remvoed it from count record cursor.